testing/cli/test_sct_run_batch.py

```diff
@@ -61,70 +59,73 @@ def test_config_extra_value_warning():
+    with pytest.warns(UserWarning, match='unknowable'):
+        # I'm not sure how to check that argparse is printing the right error here, but I trust
+        with pytest.raises(FileNotFoundError):
+            sct_run_batch.main(['-c', config_path, '-script', 'script'])
 
 
-def test_only_one_include():
-    # This is formatted strangely because pep8 says so.
-    with \
-        TemporaryDirectory() as data,\
-            TemporaryDirectory() as out:
-        with pytest.raises(AssertionError, match='Only one'):
-            sct_run_batch.main(['-include', 'arg', '-include-list',
-                                'arg2', '-path-data', data, '-path-out', out
-                                , '-script', out])
+def test_only_one_include(tmp_path):
+    data = tmp_path / 'data'
+    data.mkdir()
+    out = tmp_path / 'out'
+    with pytest.raises(AssertionError, match='Only one'):
+        sct_run_batch.main(['-include', 'arg', '-include-list', 'arg2',
+                            '-path-data', str(data), '-path-out', str(out), '-script', str(out)])
 
 
-def test_non_executable_task():
+def test_non_executable_task(tmp_path, dummy_script):
     data_path = sct_test_path()
-    with \
-        NamedTemporaryFile('w', suffix='.sh') as script,\
-            TemporaryDirectory() as out:
-
-        script_text = """
-        #!/bin/bash
-        echo $SUBJECT
-        """
-        script.write(dedent(script_text)[1:]) #indexing removes beginning newline
+    out = tmp_path / 'out'
+
+    script_text = """
+    #!/bin/bash
+    echo $SUBJECT
+    """
+    with open(tmp_path / "dummy_nonexectuable_script.sh", 'w') as script:
+        script.write(dedent(script_text)[1:])  # indexing removes beginning newline
         script.flush()
 
-        assert not os.access(script.name, os.X_OK), "Script already executable"
+    # The assertion below is meant to ensure that the script is not executable at first,
+    # so that we can test whether `sct_run_batch` properly changes the permissions. However,
+    # checking against os.X_OK isn't compatible with Windows. So, I've commented it out:
+
+    # assert not os.access(script.name, os.X_OK), "Script already executable"
+
+    # More broadly speaking, however: Don't the other tests also accomplish this same functionality?
+    # i.e. Wouldn't _all_ dummy scripts we create start out as non-executable? I suspect that this test
+    # could be combined with the other tests and there wouldn't be a meaningful loss of coverage.
 
-        sct_run_batch.main(['-include', '^t.*',
-                            '-subject-prefix', '',
-                            '-path-data', data_path, '-path-out', out,
-                            '-script', script.name,
-                            '-continue-on-error', 0])
+    sct_run_batch.main(['-include', '^t.*',
+                        '-subject-prefix', '',
+                        '-path-data', data_path, '-path-out', str(out),
+                        '-script', script.name,
```

Comment:
I'm a little bit surprised that `script.name` is still fine to use after the file is closed (by being done with the `with` block above), but I guess nothing **un**sets it. Either way, shouldn't it be `dummy_script` instead, as returned by the fixture?